javalib/src/main/scala/java/net/InetAddress.scala

```diff
@@ -216,60 +218,32 @@ object InetAddress {
      * to leave the host field blank/empty.
      */
     val effectiveHost = if (isNumeric) null else host
-
-    if (addrinfoP.ai_family == AF_INET) {
-      new Inet4Address(addrinfoToByteArray(addrinfoP), effectiveHost)
-    } else if (addrinfoP.ai_family == AF_INET6) {
+    val ipAddress = SocketHelpers.sockaddrToByteArray(addrinfoP.ai_addr)
+    if (ipAddress.length == 4) {
+      new Inet4Address(ipAddress, effectiveHost)
+    } else {
```

Comment:
at this point addrinfoP.ai_family _should_ be AF_INET6 but
there is, I believe, no guarantee that such is true.

Reply:
`sockaddrToByteArray` already check this and would have raised an exception

Reply:
Fair point & agreed.